path_roles.go

```diff
@@ -274,8 +294,21 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		return logical.ErrorResponse("either Azure role definitions, group definitions, or an Application Object ID must be provided"), nil
 	}
 
-	// save role
-	err = saveRole(ctx, req.Storage, role, name)
+	var r *roleEntry
+	var creds *ClientCredentials
+	if role.ApplicationType == "static" {
+		creds, err = b.createStaticSPSecret(ctx, client, role)
```

Comment:
Does any update to a role warrant creating a new set of credentials? Unless I missed something, `createStaticSPSecret()` and `addAppPassword()` do not appear to cull/rotate old credentials.

Reply:
oh, very good catch! only create should add the credentials, not updates to the role. And you're correct that there is no support for rotating old credentials